app/grandchallenge/reader_studies/views.py

```diff
@@ -869,17 +869,16 @@ def check_permissions(self, request):
     def get_queryset(self):
         raise NotImplementedError
 
-    def delete(self, request, *args, **kwargs):
-        objects = self.check_permissions(request)
+    def form_valid(self, *args, **kwargs):
+        objects = self.check_permissions(self.request)
         objects.delete()
-        messages.add_message(request, messages.SUCCESS, self.success_message)
-        return HttpResponse(
-            self.get_success_url(),
-            headers={
-                "HX-Redirect": self.get_success_url(),
-                "HX-Refresh": True,
-            },
-        )
+
+        response = super().form_valid(*args, **kwargs)
+
+        response["HX-Redirect"] = self.get_success_url()
+        response["HX-Refresh"] = True
+
+        return response
```

Comment:
This doesn't work properly. For the `HX-Redirect` header to work properly you need to use a HttpResponse instead of the HtpRedirectResponse from Django. Alternatively, if you want to stick to using HttpRedirectResponse, you need to remove the `Location` header as well as the `HX-Redirect` response header. 

And it needs to be `response["HX-Refresh"] = "true"` not `response["HX-Refresh"] = True`

Reply:
OK both issues should be fixed, could you test them again to verify?

Reply:
All views work as expected!